frb_example/pure_dart/dart/lib/main.dart

```diff
@@ -34,6 +34,11 @@ void main(List<String> args) async {
     expect(
         await api.primitiveTypes(myI32: 123, myI64: 10000000000000, myF64: 12345678901234567890.123, myBool: true), 42);
   });
+
+  test('dart call optional primitiveTypes', () async {
+    expect(await api.primitiveOptionalTypes(myI32: 123, myI64: null, myF64: null, myBool: null), 42);
```

Comment:
would be great to test two cases: null and non-null

```suggestion
    expect(await api.primitiveOptionalTypes(myI32: null, myI64: null, myF64: null, myBool: null), 0);
    expect(await api.primitiveOptionalTypes(myI32: 123, myI64: 123, myF64: 123, myBool: bool), 4);
```

And, would be great to modify api.rs, such that (e.g.) it returns the number of non-null values it sees. Thus we are more confident about the feature :)